platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/HttpRequest.java

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.mapboxsdk.http;
+
+import android.support.annotation.Nullable;
+
+public abstract class HttpRequest {
+
+  private static HttpRequest httpRequest;
+
+  static final int CONNECTION_ERROR = 0;
+  static final int TEMPORARY_ERROR = 1;
+  static final int PERMANENT_ERROR = 2;
+
+  public static synchronized void setHttpRequest(@Nullable HttpRequest requestImpl) {
+    httpRequest = requestImpl;
+  }
+
+  public abstract void executeRequest(NativeHttpRequest httpRequest, long nativePtr, String resourceUrl,
+                                      String etag, String modified);
+
+  public abstract void cancelRequest(long nativePtr);
+
+  static void execute(NativeHttpRequest httpRequest, long nativePtr, String resourceUrl,
+                             String etag, String modified) {
+    getInstance().executeRequest(httpRequest, nativePtr, resourceUrl, etag, modified);
+  }
+
+  static void cancel(long nativePtr) {
+    getInstance().cancelRequest(nativePtr);
+  }
+
+  private static synchronized HttpRequest getInstance() {
+    if (httpRequest == null) {
+      httpRequest = new OkHttpRequest();
```

Comment:
Part of the goal of having modules is that OkHttp does not need to be compiled into the APK. I think there should be a `DefaultModules.java` file that sets all the default modules. Then for someone who wants to have different modules, they just leave DefaultModules out of the compilation and make their own `MyModules`.

Reply:
I'm aware of two techniques to remove code from an aar/binary:
 - gradle exclusion
 - proguard

Please elaborate how `they just leave DefaultModules out of the compilation and make their own MyModules`.

Reply:
I was thinking an option in the build rules to leave `DefaultModules.java` out - so I think that would be `gradle exclusion` though I'm not actually familiar with gradle.